templates/taxi/driver_list.html

```diff
@@ -7,7 +7,12 @@ <h1>
         Create
       </a>
     </h1>
-
+    <form method="get" action="" class="mb-4">
+      <label>
+        <input type="text" name="username" class="form-control" placeholder="Search by username" value="{{ request.GET.username }}">
+        <input class="btn btn-secondary" type="submit" value="🔎">
+      </label>
+    </form>
```

Comment:
Create a separate template for search form and include it where you need to avoid code duplication